### PR TITLE
refactor: internal refactor for prepare new arch

### PR DESCRIPTION
**Pull request by YangJonghun**
## Summary
refactor: internal refactor for prepare new arch

## Test plan

### CODE REVIEW

File: src/types/ViewType.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@
  * Define Available view type for android
  * these values shall match android spec, see ViewType.kt
  */
-enum ResizeMode {
+enum ViewType {
```
Comment (freeboub) [inline src/types/ViewType.ts:5]: good catch 👍

**Review by freeboub — APPROVED** [review]

**Comment by freeboub** [comment]
need to change target brach !

**Comment by YangJonghun** [comment]
@freeboub 
This PR is written so that it can be merged into master first.
I've just pre-named or pre-processed it to make easier to write this lib for new arch.

**Review by KrzysztofMoch — APPROVED** [review]
@freeboub I agree with @YangJonghun this PR mainly refactor names and types - it does not introduce any logic/behaviour change for user, so we can merge this into master. Also it will be easier to sync this later

**Comment by freeboub** [comment]
Ok for me, let's go !
Sorry for my mow availability these times..

**Review by freeboub — APPROVED** [review]

**Comment by YangJonghun** [comment]
@freeboub @KrzysztofMoch 
PR #3962 may cause problem with users who use 0.73.x (0.73.0's default Kotlin version is 1.9.0)

So, I add below commit within this PR.
https://github.com/TheWidlarzGroup/react-native-video/pull/3980/commits/3f9628429db00ae582c07d81a4e7cce8a1091a14

related issue : https://github.com/TheWidlarzGroup/react-native-video/issues/3998

**Comment by KrzysztofMoch** [comment]
Yeah I saw it - Thanks! I merged this and I will do `6.4.1` release

**Pull request merged**